samples/bluetooth/direction_finding_central/src/main.c

```diff
@@ -0,0 +1,278 @@
+#endif /* CONFIG_BT_CTLR_DF_ANT_SWITCH_RX */
+	};
+
+	const struct bt_df_conn_cte_req_params cte_req_params = {
+		.interval = CTE_REQ_INTERVAL,
+		.cte_length = CTE_LEN,
+#if defined(CONFIG_BT_CTLR_DF_ANT_SWITCH_RX)
+		.cte_type = BT_DF_CTE_TYPE_AOA,
+#else
+		.cte_type = BT_DF_CTE_TYPE_AOD_2US,
+#endif /* CONFIG_BT_CTLR_DF_ANT_SWITCH_TX */
+	};
+
+	printk("Enable receiving of CTE...\n");
+	err = bt_df_conn_cte_rx_enable(default_conn, &cte_rx_params);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return;
+	}
+	printk("success. CTE receive enabled.\n");
+
+	printk("Request CTE from peer device...\n");
+	err = bt_df_conn_cte_req_enable(default_conn, &cte_req_params);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return;
+	}
+	printk("success. CTE request enabled.\n");
+}
+
+static void start_scan(void)
+{
+	int err;
+
+	/* Use active scanning and disable duplicate filtering to handle any
+	 * devices that might update their advertising data at runtime.
+	 */
+	struct bt_le_scan_param scan_param = {
+		.type = BT_LE_SCAN_TYPE_ACTIVE,
+		.options = BT_LE_SCAN_OPT_NONE,
+		.interval = BT_GAP_SCAN_FAST_INTERVAL,
+		.window = BT_GAP_SCAN_FAST_WINDOW,
+	};
+
+	err = bt_le_scan_start(&scan_param, device_found);
+	if (err) {
+		printk("Scanning failed to start (err %d)\n", err);
+		return;
+	}
+
+	printk("Scanning successfully started\n");
+}
+#pragma GCC pop_options
+
+static void connected(struct bt_conn *conn, uint8_t conn_err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+	int err;
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (conn_err) {
+		printk("Failed to connect to %s (%u)\n", addr, conn_err);
+
+		bt_conn_unref(default_conn);
+		default_conn = NULL;
+
+		start_scan();
+		return;
+	}
+
+	printk("Connected: %s\n", addr);
+
+	if (conn == default_conn) {
+		enable_cte_reqest();
+	}
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Disconnected: %s (reason 0x%02x)\n", addr, reason);
+
+	if (default_conn != conn) {
+		return;
+	}
+
+	bt_conn_unref(default_conn);
+	default_conn = NULL;
+
+	start_scan();
+}
+
+static void cte_recv_cb(struct bt_conn *conn, struct bt_df_conn_iq_samples_report const *report)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (report->err == BT_DF_IQ_REPORT_ERR_SUCCESS) {
+		printk("CTE[%s]: samples count %d, cte type %s, slot durations: %u [us], "
+		       "packet status %s, RSSI %i\n",
+		       addr, report->sample_count, cte_type2str(report->cte_type),
+		       report->slot_durations, packet_status2str(report->packet_status),
+		       report->rssi);
+	} else {
+		printk("CTE[%s]: request failed, err %u\n", addr, report->err);
+	}
+}
+
+static struct bt_conn_cb conn_callbacks = {
+	.connected = connected,
+	.disconnected = disconnected,
+	.cte_report_cb = cte_recv_cb,
+};
```

Comment:
Consider using the new static CONN_CB_DEFINE (or whatever it is called), as I think that is possibly the suggested way of doing that now

Reply:
Thanks I didn't know about `BT_CONN_CD_DEFINE`.